src/Functions/FunctionsStringRegex.cpp

```diff
@@ -1081,6 +1082,148 @@ class FunctionStringReplace : public IFunction
     }
 };
 
+/** Match all groups of given input string with given re, return array of arrays of matches.
+ *
+ *  SELECT extractAllGroups('abc=111, def=222, ghi=333', '("[^"]+"|\\w+)=("[^"]+"|\\w+)')
```

Comment:
If we have `extractAllGroups`, it's reasonable to already have `extractGroups` function.